src/main/java/org/opentripplanner/model/plan/LegTime.java

```diff
@@ -0,0 +1,43 @@
+package org.opentripplanner.model.plan;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * A scheduled time of a transit vehicle at a certain location with a optional realtime information.
+ */
+public record LegTime(@Nonnull ZonedDateTime scheduledTime, @Nullable RealTimeEstimate estimated) {
+  public LegTime {
+    Objects.requireNonNull(scheduledTime);
+  }
+  @Nonnull
+  public static LegTime of(ZonedDateTime realtime, int delaySecs) {
+    var delay = Duration.ofSeconds(delaySecs);
+    return new LegTime(realtime.minus(delay), new RealTimeEstimate(realtime, delay));
```

Comment:
Is the delay always accurate difference between scheduled and realtime times?

Reply:
It must be otherwise we have a huge problem in the system. Or what do you mean?

Reply:
I was just wondering if there is some rounding somewhere to 30 seconds for example or random bugs with frequencies/flex etc.